hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -1730,4 +1750,39 @@ private static boolean isReadCall(Method method) {
     }
     return !method.getAnnotationsByType(ReadOnly.class)[0].activeOnly();
   }
+
+  /**
+   * Checks and sets last refresh time for a namespace's stateId.
+   * Returns true if refresh time is newer than threshold.
+   * Otherwise, return false and call should be handled by active namenode.
+   * @param nsId namespaceID
+   */
+  @VisibleForTesting
+  boolean isNamespaceStateIdFresh(String nsId) {
+    if (activeNNStateIdRefreshPeriodMs < 0) {
+      LOG.debug("Skipping freshness check and returning True since"
```

Comment:
Move this to the initialization, so that we only get the log once rather than every call.

Reply:
Actually, you should also use expansion pattern rather than string concatenation.

LOG.debug("Skipping ... {} is less than zero.", RBFConfigKeys....);